website/docs/reference/dbt-jinja-functions/statement-blocks.md

```diff
@@ -31,6 +33,35 @@ The signature of the `statement` block looks like this:
 statement(name=None, fetch_result=False, auto_begin=True)
 ```
 
+When executing a `statement`, dbt needs to understand how to resolve references to other dbt models or resources. You can explicitly specify it using `-- depends_on` or doing a `ref()` function outside the statement block to ensure proper execution.
+
+<expandable alt_header="Example using -- depends_on">
+
+```sql
+-- depends_on: {{ ref('users') }}
+
+{%- call statement('states', fetch_result=True) -%}
+
+    select distinct state from {{ ref('users') }}
+
+{%- endcall -%}
```

Comment:
```suggestion
{% call statement('states', fetch_result=True) -%}

    select distinct state from {{ ref('users') }}

    /*
    The unique states are: {{ load_result('states')['data'] }}
    */
{%- endcall %}
```

Suggest a bit less whitespace control to respect any other newlines in the user's code, and demonstrate how to interact with the result of the query